allocator/ascendalloc/ascendalloc.go

```diff
@@ -1,96 +1,50 @@
-// Package ascendalloc implements an ipfscluster.Allocator returns allocations
-// based on sorting the metrics in ascending order. Thus, peers with smallest
-// metrics are first in the list. This allocator can be used with a number
-// of informers, as long as they provide a numeric metric value.
+// Package ascendalloc implements an ipfscluster.allocator.Allocator returns
+// allocations based on sorting the metrics in ascending order. Thus, peers with
+// smallest metrics are first in the list. This allocator can be used with a
+// number of informers, as long as they provide a numeric metric value.
 package ascendalloc
 
 import (
 	"sort"
-	"strconv"
 
+	"github.com/ipfs/ipfs-cluster/allocator"
 	"github.com/ipfs/ipfs-cluster/api"
 
-	rpc "github.com/hsanjuan/go-libp2p-gorpc"
 	cid "github.com/ipfs/go-cid"
 	logging "github.com/ipfs/go-log"
 	peer "github.com/libp2p/go-libp2p-peer"
 )
 
 var logger = logging.Logger("ascendalloc")
 
-// Allocator implements ipfscluster.Allocate.
-type Allocator struct{}
-
-// NewAllocator returns an initialized Allocator
-func NewAllocator() *Allocator {
-	return &Allocator{}
+// AscendAllocator extends the SimpleAllocator
```

Comment:
SimpleAllocator does nothing so I don't quite get the benefit of extending. I think it is simpler to implement the ipfscluster.PinAllocator interface directly and use the common MetricSorter.